webrtc.html

```diff
@@ -2582,6 +2636,110 @@
+          [[!WebCryptoAPI]] <a href="https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#dfn-AlgorithmIdentifier">AlgorithmIdentifier</a>
+          type.  The <var>keygenAlgorithm</var> value MUST be a valid argument
+            to <a href="https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#SubtleCrypto-method-generateKey"><code>Crypto.subtle.generateKey</code></a>.</p>
+
+          <p>The selected <code>AlgorithmIdentifier</code> MUST be valid for
+          both key generation and signing.  The following values are valid:
+            <a href="https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#rsassa-pkcs1">RSASSA-PKCS1-v1_5</a>,
+            <a href="https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#rsa-pss">RSA-PSS</a>,
+            <!-- (note, PSS is not useful with current TLS but is included for future-proofing), -->
+            and <a href="https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#ecdsa">ECDSA</a>.</p>
+
+          <p>The resulting certificate contains a signature.  The validity of
+          this signature is only relevant for compatibility reasons.  Only the
+          public key and the resulting certificate fingerprint are used for
+          WebRTC, but it is more likely that the certificate will be accepted if
+          the certificate is well formed.  The browser selects the algorithm
+          used to sign the certificate; a browser SHOULD select SHA-256 if a
+          hash algorithm is needed.</p>
+
+          <p>The resulting certificate MUST NOT include information that can be
+          linked to a user or <a>user agent</a>.  Randomized values for
+          distinguished name and serial number SHOULD be used.</p>
+
+          <p>A <a>user agent</a> MUST reject a call
+          to <code>generateCertificate()</code> that uses an algorithm that is
+          not supported: either because the algorithms are not supported or it
+          cannot use a certificate produced by those algorithms
+          with <code>RTCPeerConnection</code>.</p>
+        </dd>
+      </dl>
+
+      <section>
+        <h2>RTCCertificate Interface</h2>
+
+        <p>The <dfn><code>RTCCertificate</code></dfn> interface contains no
+        attributes or methods.  However, internal slots contain a handle to the
+        generated private keying materal
+        ([[<dfn id="RTCCertificate-handle">handle</dfn>]]) and a certificate
+        ([[<dfn id="RTCCertificate-certificate">certificate</dfn>]])
+        that <code>RTCPeerConnection</code> uses to authenticate with a
+        peer.</p>
+
+        <p>For the purposes of this API, the
+        [[<a href="RTCCertificate-certificate">certificate</a>]] slot contains
+        unstructured binary data.</p>
+
+        <p>Note that a <code>RTCCertificate</code> might not directly hold
+        private keying material, this might be stored in a secure module.</p>
```

Comment:
Does this mean that if the certificate is persisted, the persisted object may have links to something that is not part of the persistent storage?
Do we have some kind of assurance that if persistent storage is cleared, the linked-to objects in the secure module get cleared out too? This makes me nervous about relative lifetimes and well-defined failure modes.